src/Sulu/Bundle/SecurityBundle/Controller/ProfileController.php

```diff
@@ -10,20 +10,161 @@
+     * @param ObjectManager $objectManager
+     * @param ViewHandlerInterface $viewHandler
+     * @param UserSettingRepositoryInterface $userSettingRepository
+     */
+    public function __construct(
+        TokenStorageInterface $tokenStorage,
+        ObjectManager $objectManager,
+        ViewHandlerInterface $viewHandler,
+        UserSettingRepositoryInterface $userSettingRepository
+    ) {
+        $this->tokenStorage = $tokenStorage;
+        $this->objectManager = $objectManager;
+        $this->viewHandler = $viewHandler;
+        $this->userSettingRepository = $userSettingRepository;
+    }
+
+    /**
+     * Sets the given language on the current user
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function putLanguageAction(Request $request)
     {
-        $user = $this->getUser();
+        $user = $this->tokenStorage->getToken()->getUser();
         $user->setLocale($request->get('locale'));
 
-        $em = $this->getDoctrine()->getManager();
-        $em->flush();
+        $this->objectManager->flush();
+
+        return $this->viewHandler->handle(View::create(array('locale' => $user->getLocale())));
+    }
+
+    /**
+     * Takes a key, value pair and stores it as settings for the user
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function putSettingsAction(Request $request)
+    {
+        $key = $request->get('key');
+        $value = $request->get('value');
+
+        try {
+            if (!$key) {
+                throw new MissingArgumentException(static::$entityNameUserSetting, 'key');
+            }
+
+            if (!$value) {
+                throw new MissingArgumentException(static::$entityNameUserSetting, 'value');
+            }
+
+            $user = $this->tokenStorage->getToken()->getUser();
+
+            // encode before persist
+            $data = json_encode($value);
+
+            // get setting
+            // TODO: move this logic into own service (UserSettingManager?)
+            $setting = $this->userSettingRepository->findOneBy(array('user' => $user, 'key' => $key));
+
+            // or create new one
+            if (!$setting) {
+                $setting = new UserSetting();
+                $setting->setKey($key);
+                $setting->setUser($user);
+                $this->objectManager->persist($setting);
+            }
+
+            // persist setting
+            $setting->setValue($data);
+            $this->objectManager->flush();
+
+            //create view
+            $view = View::create($setting, 200);
+        } catch (RestException $exc) {
+            $view = View::create($exc->toArray(), 400);
+        }
+
+        return $this->viewHandler->handle($view);
+    }
+
+    /**
+     * Returns the settings for a key for the current user
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function getSettingsAction(Request $request)
```

Comment:
@chirimoya @wachterjohannes @dantleech Can't find any usage of this action, if a user setting is set using the javascript extension it is set first on the client and then sent to the server, and loaded via the `SULU` javascript object over the controller which renders the index file.

Not sure if I should remove it, or leave it here just for completeness. Not saying what I would slightly prefer, since I want to hear your honest opinions :wink: